src/main/java/fr/techad/edc/httpd/SearchHandler.java

```diff
@@ -16,36 +18,55 @@
 import java.util.Map;
 
 /**
- * TECH ADVANTAGE
- * All right reserved
- * Created by cochon on 24/04/2018.
+ * TECH ADVANTAGE All right reserved Created by cochon on 24/04/2018.
  */
 public class SearchHandler implements HttpHandler {
-    static final Logger LOGGER = LoggerFactory.getLogger(SearchHandler.class);
-    private final ObjectMapper objectMapper;
-    private final WebServerConfig config;
+  static final Logger LOGGER = LoggerFactory.getLogger(SearchHandler.class);
+  private final ObjectMapper objectMapper;
+  private final WebServerConfig config;
+
+  public SearchHandler(ObjectMapper objectMapper, WebServerConfig config) {
+    this.objectMapper = objectMapper;
+    this.config = config;
+  }
+
+  public void handleRequest(HttpServerExchange exchange) throws Exception {
 
-    public SearchHandler(ObjectMapper objectMapper, WebServerConfig config) {
-        this.objectMapper = objectMapper;
-        this.config = config;
+    Map<String, Deque<String>> queryParameters = exchange.getQueryParameters();
+    LOGGER.debug("Query Parameters: {}, Query: {}", queryParameters, exchange.getQueryString());
+
+    Deque<String> query = queryParameters.get("query");
+    Deque<String> language = queryParameters.get("lang");
+    Deque<String> limit = queryParameters.get("limit");
+    Deque<String> strict = queryParameters.get("strict");
+    Boolean strictMode = false;
+    String lang = "";
+    int limitNumber = 100;
+    if (strict != null && language != null && limit != null) {
```

Comment:
Each parameter is independent og the others. The user can set a limit to 20 results without defining the strict mode. By the way, strict should be renamed to exact (in url and here)